napari/_vispy/overlays/brush_circle.py

```diff
@@ -36,6 +36,10 @@ def __init__(self, *, viewer, overlay, parent=None):
 
         self.reset()
 
+        # manually trigger _on_canvas_change once and get the correct canvas
+        self.node.parent = None
+        self.node.parent = parent
```

Comment:
What do you think about this, rather than going through the events?
```
parent.scene.canvas.events.mouse_move.connect(self._on_mouse_move)
```

Reply:
Of course, much better!